modules/opsgenie-team/modules/integration/main.tf

```
@@ -0,0 +1,194 @@
+  namespace   = ""
+  tenant      = ""
+  environment = ""
+  stage       = ""
+
+  context = module.this.context
+}
+
+data "opsgenie_team" "default" {
+  count = local.enabled ? 1 : 0
+
+  name = var.team_name
+}
+
+module "api_integration" {
+  source  = "cloudposse/incident-management/opsgenie//modules/api_integration"
+  version = "0.15.0"
+
+  # TODO: add additional parameters to api integrations
+  api_integration = {
+    name          = local.name
+    type          = var.type
+    owner_team_id = local.team_id
+  }
+
+  context = module.this.context
+}
+
+resource "opsgenie_integration_action" "datadog" {
+  # TODO: use upstream module instead of raw resource
+  # source  = "cloudposse/incident-management/opsgenie//modules/integration_action"
+  # version = "0.14.4"
+
+  count = local.append_datadog_tags_enabled ? 1 : 0
+
+  integration_id = module.api_integration.api_integration_id
+
+  create {
+    ignore_responders_from_payload = true
+
+    name          = "Create Metric Alert"
+    alias         = "{{alias}}"
+    tags          = ["{{dd_tags}}"]
+    user          = "Datadog"
+    note          = "{{note}}"
+    source        = "{{source}}"
+    message       = "[Datadog] {{monitor_name}}"
+    entity        = "{{entity}}"
+    alert_actions = []
+
+    # The Datadog description is null so we use the DD Message as the description
+    description = "{{templated_message}}\n"
+    extra_properties = {
+      "Event Url" : "{{event_url}}"
+      "Snapshot Link" : "{{snapshot_link}}"
+      "Metric Graph" : "<img src=\"{{snapshot_url}}\"/>"
+      "Datadog Tags" : "{{dd_tags}}"
+    }
+
+    # Note, this does NOT match the docs found in terraform
+    ## https://registry.terraform.io/providers/opsgenie/opsgenie/latest/docs/resources/integration_action#filter
+    # Instead we find the actual working fields from the debug section of Opsgenie.
+    filter {
+      type = "match-all-conditions"
+
+      # The order of these filter rules matter or it can create an inconsistent plan. It seems like it's alphabetically
+      # ordered by the conditions field.
+
+      conditions {
+        # not `actions` as described in the TF docs
+        field          = "action"
+        operation      = "equals"
+        expected_value = "create"
+      }
+
+      # Specifically do not check for dd_tags, contains, team:${var.team_name} so other teams can reuse a team integration. Also @opsgenie-<team> already controls what integration the alert is sent to.
+
+      # We purposely do not check for tenant/stage since this is controlled by the Datadog organizations
+      # and how the datadog organizations are configured. 1 per tenant, 1 per account, single org, etc.
+
+      conditions {
+        # not `source` as described in the TF docs
+        field          = "source_type_name"
+        operation      = "equals"
+        expected_value = "Monitor Alert"
+      }
```

Do not create Datadog Synthetics when only monitors are to be created.

```suggestion
      conditions {
        # not `source` as described in the TF docs
        field          = "event_type"
        operation      = "equals"
        expected_value = "query_alert_monitor"
      }
```